### PR TITLE
273 - Editing an experiment tag leads to blank page

### LINKED ISSUE #273: Editing an experiment tag leads to blank page
By editing an already existing experiment and trying to add a tag, as soon as the user clicks on the tag field it is redirected to a blank page with the following errors in the console

https://github.com/MetaCell/salk-interactive-atlas/assets/99416933/f88c995c-a254-4b6d-ae18-24e70e61e5c8

Steps to Reproduce:
1. Login
2. Create an experiment (or having an already created experiment)
3. Click on top of the experiment to open the submenu
4. Click on the Edit info and tags 
5. Add a new tag

### PR DESCRIPTION
**Pull request by Aiga115**
Closes #273 

Implemented solution:  
Displayed dialog window on card level, implemented on change function for each instance (name, desciption, tags), which changes the state, after calling the updateExperiment api we change the experiment then we call the function to call the experiments list again

### CODE REVIEW

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
Looks good. There's only one small problem with the context menu changing position sometimes:

https://github.com/MetaCell/salk-interactive-atlas/assets/19196034/951d5bee-438f-4245-a9a9-42b886480695

If it's easier, I would be ok with the context menu being closed on modal close

File: applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx [hunk 1]
```diff
@@ -36,35 +27,106 @@ const useStyles = makeStyles(() => ({
...
   return (
     <Modal
       dialogActions={true}
-      actionText="Create"
+      actionText="Edit"
       disableGutter={true}
       open={open}
       handleClose={handleClose}
+      handleAction={handleAction}
       title="Exploration of the spinal cord"
```
Comment (afonsobspinto) [inline applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx:108]: This title doesn't make much sense, can you check with @neuroBazzi or @syamace-metacell what they recommend to use?

File: applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx [hunk 2]
```diff
@@ -36,35 +27,106 @@ const useStyles = makeStyles(() => ({
...
+  const addTags = async (tags: string[]) => {
+    try {
+      await api.addTagsExperiment(experiment.id.toString(), tags);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  const handleAction = async () => {
+    const deletedTags = initialTags.filter((tag: any) => !tags.includes(tag));
```
Comment (afonsobspinto) [inline applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx:82]: The action should only occur if there are no errors in the form. Aka, all the required fields are filled.
Name and description are required fields

File: applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx (hunk 1 above)
Comment (syamace-metacell) [reply applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx:108]: @afonsobspinto Based on the designs, the title is supposed to be the current experiment's name, the one that the user is currently editing it's info & tags. So in this case (the video above), "Dataset V3 - Edited" should be the title. cc @neuroBazzi

File: applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx [hunk 3]
```diff
@@ -78,7 +98,41 @@ export const ExplorationSpinalCordDialog = (props: any) => {
...
+      }
+    }
+
+    return errorsSet
+  }
+
   const handleAction = async () => {
+    setIsLoading(true)
+    const errorsSet = await getValidationErrors()
+    console.log("errorset: ", errorsSet)
```
Comment (afonsobspinto) [inline applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx:117]: This can be deleted

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
Trying to edit a experiment twice (in sequence) doesn't seem to be working:
https://github.com/MetaCell/salk-interactive-atlas/assets/19196034/1d107be8-40ec-458b-9c26-df993a4727b2

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
I also noticed that your fix of the merge conflict is adding back some files that were removed purposely (namely data/residential_populations and objs files) which is not correct. Might be easier to branch from develop again and add your commits on top of that

File: applications/portal/frontend/src/components/common/ExperimentDialogs/TagsAutocomplete.jsx [hunk 4]
```diff
@@ -34,18 +34,18 @@ export const TagsAutocomplete = (props) => {
...
-      options={tags}
-      defaultValue={[]}
+      value={tags}
+      options={tagsOptions}
       freeSolo
       limitTags={3}
       renderTags={(value, getTagProps) =>
         value.map((option, index) => (
-          <Chip key={index} style={{...tagsColorOptions[index]}} onDelete={() => console.log(option)} label={option} {...getTagProps({ index })} />
+          <Chip key={index} style={{ ...tagsColorOptions[index] }} onDelete={() => console.log(option)} label={option} {...getTagProps({ index })} />
```
Comment (afonsobspinto) [inline applications/portal/frontend/src/components/common/ExperimentDialogs/TagsAutocomplete.jsx:43]: This onDelete callback looks strange

File: applications/portal/frontend/src/App.tsx [hunk 5]
```diff
@@ -89,7 +89,7 @@ export const App = (props: any) => {
                 {loading ? <Loader/> : (
                     <>
                       <ProtectedRoute exact={true} path="/">
-                        <HomePage latestExperimentId={latestExperimentId} />
+                        <HomePage latestExperimentId={latestExperimentId} onExperimentChange={(id: string) => setLatestExperimentId(id)} />
```
Comment (afonsobspinto) [inline applications/portal/frontend/src/App.tsx:92]: My guess is that the onExperimentChange callback is the reason why the bug I reported is happening. 
When we change an experiment twitce, on the second change the id of the latest experiment doesn't change therefore there's no re-render of the dashboard.

**Review by afonsobspinto — CHANGES_REQUESTED** [review]

**Comment by afonsobspinto** [comment]
replaced by https://github.com/MetaCell/salk-interactive-atlas/pull/317

**Pull request closed without merging**